packages/core/src/NodeExecuteFunctions.ts

```diff
@@ -1647,6 +1652,187 @@ function cleanupParameterData(inputData: NodeParameterValueType): NodeParameterV
+		);
+	}
+	if (extracted.length < 2 || extracted.length > 2) {
+		throw new Error(
+			`Property "${parameterName}" has an invalid extractValue regex "${regex.source}". extractValue expects exactly one group to be returned.`,
+		);
+	}
+	return extracted[1];
+}
+
+function extractValueRLC(
+	value: NodeParameterValueType | object,
+	property: INodeProperties,
+	parameterName: string,
+	nodeType: INodeType,
+): NodeParameterValueType | object {
+	// Not an RLC value
+	if (typeof value !== 'object' || !value || !('mode' in value) || !('value' in value)) {
+		return value;
+	}
+	const modeProp = findPropertyInArray(value.mode as string, property.modes || []);
+	if (!modeProp) {
+		return value.value;
+	}
+	if (!('extractValue' in modeProp) || !modeProp.extractValue) {
+		return value.value;
+	}
+
+	if (typeof value.value !== 'string') {
+		let typeName: string | undefined = value.value?.constructor.name;
+		if (value.value === null) {
+			typeName = 'null';
+		} else if (typeName === undefined) {
+			typeName = 'undefined';
+		}
+		throw new Error(
+			`Only strings can be passed to extractValue. Parameter "${parameterName}" passed "${typeName}"`,
+		);
+	}
+
+	if (modeProp.extractValue.type !== 'regex') {
+		throw new Error(
+			`Property "${parameterName}" has an unknown extractValue type "${
+				modeProp.extractValue.type as string
+			}"`,
+		);
+	}
+
+	const regex = new RegExp(modeProp.extractValue.regex);
+	return executeRegexExtractValue(value.value, regex, parameterName);
+}
+
+function extractValueOther(
+	value: NodeParameterValueType | object,
+	property: INodeProperties | INodePropertyCollection,
+	parameterName: string,
+	nodeType: INodeType,
+): NodeParameterValueType | object {
+	if (!('extractValue' in property) || !property.extractValue) {
+		return value;
+	}
+
+	if (typeof value !== 'string') {
+		let typeName: string | undefined = value?.constructor.name;
+		if (value === null) {
+			typeName = 'null';
+		} else if (typeName === undefined) {
+			typeName = 'undefined';
+		}
+		throw new Error(
+			`Only strings can be passed to extractValue. Parameter "${parameterName}" passed "${typeName}"`,
+		);
+	}
+
+	if (property.extractValue.type !== 'regex') {
+		throw new Error(
+			`Property "${parameterName}" has an unknown extractValue type "${
+				property.extractValue.type as string
+			}"`,
+		);
+	}
+
+	const regex = new RegExp(property.extractValue.regex);
+	return executeRegexExtractValue(value, regex, parameterName);
+}
+
+/**
+ * Extracts wanted value from a provided value using a property's extractor.
+ *
+ * @param {(NodeParameterValue | INodeParameters | NodeParameterValue[] | INodeParameters[] | object)} value
+ * @param {string} parameterName
+ * @param {INode} node
+ * @param {INodeType} nodeType
+ * @returns {(NodeParameterValue | INodeParameters | NodeParameterValue[] | INodeParameters[] | object)}
```

Comment:
JSDoc adds nothing not already covered by TS type signature. Remove?

Reply:
Can do. I was just following the convention for the file.